src/utils.ts

```diff
@@ -326,3 +319,20 @@ export function withAndMaybeCheckIsNotNull(options: Pick<Options, "useNullAsOpti
 export function withAndMaybeCheckIsNull(options: Pick<Options, "useNullAsOptional">, typeName: string) {
   return maybeCheckIsNotNull(options, typeName, "&&");
 }
+
+export async function getVersions(request: CodeGeneratorRequest) {
+  let protocVersion = "unknown";
+  if (request.compilerVersion) {
+    const { major, minor, patch } = request.compilerVersion;
+    protocVersion = `v${major}.${minor}.${patch}`;
+  }
+
+  const path: string = "../package.json";
+  const packageJson = await import(path);
+  const tsProtoVersion = `v${packageJson?.version ?? "unknown"}`;
```

Comment:
This might be a bit of a hack. When providing the path to the dynamic import directly, `tsc` will include it into the compilation, which leads to the build output having the code in the `src` sub folder. 
By moving it out, and typing it as a string, it can no longer infer that path, and then won't include it in the output.

The path is still valid on the published npm package, from running yarn pack and checking the output.

Reply:
Ah cool, good find!

I'm honestly a little surprised the `await import` just works; do we have to worry about whether ts-proto is invoked in a CJS/ESM environment? I guess this passed the build on node 16, so it must be working.

If users report issues, we could also look at doing a post-tsc step that uses `sed`/something to replace a hard-coded `const tsProtoVersion = "placeholder"` type line with the just-being-published version, in the `.js` file, before upload to npm.

But I'm good with using this and seeing how it goes!